staging/src/k8s.io/client-go/discovery/cached/disk/round_tripper.go

```diff
@@ -63,3 +66,54 @@ func (rt *cacheRoundTripper) CancelRequest(req *http.Request) {
 }
 
 func (rt *cacheRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt.Transport }
+
+// A crcDiskCache is a cache backend for github.com/gregjones/httpcache. It is
+// similar to httpcache's diskcache package, but uses checksums to ensure cache
+// integrity rather than fsyncing each cache entry in order to avoid performance
+// degradation on MacOS.
+//
+// See https://github.com/kubernetes/kubernetes/issues/110753 for more.
+type crcDiskCache struct {
+	disk *diskv.Diskv
+}
+
+// Get the requested key from the cache on disk. If Get encounters an error, or
+// the returned value is not a CRC-32 checksum followed by bytes with a matching
+// checksum it will return false to indicate a cache miss.
+func (c *crcDiskCache) Get(key string) ([]byte, bool) {
+	b, err := c.disk.Read(sanitize(key))
+	if err != nil || len(b) < binary.MaxVarintLen32 {
+		return []byte{}, false
+	}
+
+	response := b[binary.MaxVarintLen32:]
```

Comment:
Wouldn't it be simpler to just hash the bytes of the file and verify they match the key?

Reply:
(I'm not worried about performance because a cache miss means a network trip, which is going to be much slower)

Reply:
@lavalamp I think you might have misinterpreted what's happening here. The first `binary.MaxVarintLen32` of the file contains a CRC-32 checksum of the (entire) remainder of the file, which is the actual cached content.

The key `httpcache` passes to this method is the HTTP method and URL of the request it wants to lookup in the cache (e.g. `GET https://apiserver.example.org:8080/apis/v1/coolstuff`). Per my response below the only reason we hash that key is to sanitize it to ensure it's a valid filename.

Would it help to update the Godoc string or add additional commentary to make this clearer?

Reply:
@lavalamp FWIW that was my intuition too when I first looked into this. 😄